specs/WebResourceRequestedWorkers.md

```diff
@@ -0,0 +1,210 @@
+WebResourceRequested events for workers
+===
+
+# Background
+Currently [WebResourceRequested events](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/iwebview2webview5?view=webview2-0.8.355#add_webresourcerequested) are raised only for top level document HTTP requests. We are extending [AddWebResourceRequestedFilter](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/iwebview2webview5?view=webview2-0.8.355#addwebresourcerequestedfilter) API to allow you additionally to subscribe to events raised from Service or Shared workers and fixing an issue where out of process iframes did not raise WebResourceRequested events as part of the main page.
+
+# Examples
+Subscribe to WebResourceRequested event for service workers and override the image in request with the local one
+
+.Net
+```c#
+webView.CoreWebView2.AddWebResourceRequestedFilter("*.jpg", CoreWebView2WebResourceContext.All, CoreWebView2WebResourceRequestSourceKinds.ServiceWorker);
+webView.CoreWebView2.WebResourceRequested += (sender, args) =>
+{
+    if (args.RequestSourceKinds == CoreWebView2WebResourceRequestSourceKinds.ServiceWorker &&
```

Comment:
Do I need to check the SourceKinds if I used a filter when I used a filter?  Could this be an assertion?    

Do I need to use a FlagSet check instead of == equality since the input param supports Flag operations? (Not sure if there's a best practice for assuming exactly one flag set in context)

Reply:
Agreed the [flags] on this is odd here, I can't find any precedent for it.

The alternative would be to have two enums, one singular for the event args and one plural for the method.

Another alternative would be to give the _property_ a singular name:

```cs
args.RequestSourceKind == CoreWebView2WebResourceRequestSourceKinds.ServiceWorker
```

Reply:
Its a common pattern for this API to check the args because usually the end developer has more than one filter set. We should update the sample to have more than one filter set.